File: medium/container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l=0
        r=len(height)-1
        maxmax=0
        while (l<r):
            maxmax=max(maxmax,min(height[r],height[l])*(r-l))
            if height[l]<height[r]:
                l+=1
            else:
                r-=1
        return maxmax

File: medium/container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int l=0,r=height.length-1;
        int res=0;
        while (l<r) {
            res=Integer.max(res,Integer.min(height[l],height[r])*(r-l));
            if (height[r]<height[l]) {
                r--;
            } else {
                l++;
            }
        }
        return res;
    }
}
